### ISSUE TITLE
[debian-6.0.12] dpkg: error processing archive

### ISSUE THREAD

**Issue by kingorgg** [linux-surface/linux-surface#1007]
Hello.
I'm unable to upgrade the kernel to 6.0.12, from 6.0.9, on my LMDE 5 machine as I get the following error.

```cli
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following NEW packages will be installed:
  linux-headers-6.0.12-surface linux-image-6.0.12-surface
The following packages will be upgraded:
  linux-headers-surface linux-image-surface
2 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/88.8 MB of archives.
After this operation, 416 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
dpkg-deb: error: archive '/var/cache/apt/archives/linux-headers-6.0.12-surface_6.0.12-surface-1_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive /var/cache/apt/archives/linux-headers-6.0.12-surface_6.0.12-surface-1_amd64.deb (--unpack):
 dpkg-deb --control subprocess returned error exit status 2
dpkg-deb: error: archive '/var/cache/apt/archives/linux-headers-surface_6.0.12-surface-1_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive /var/cache/apt/archives/linux-headers-surface_6.0.12-surface-1_amd64.deb (--unpack):
 dpkg-deb --control subprocess returned error exit status 2
dpkg-deb: error: archive '/var/cache/apt/archives/linux-image-6.0.12-surface_6.0.12-surface-1_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive /var/cache/apt/archives/linux-image-6.0.12-surface_6.0.12-surface-1_amd64.deb (--unpack):
 dpkg-deb --control subprocess returned error exit status 2
dpkg-deb: error: archive '/var/cache/apt/archives/linux-image-surface_6.0.12-surface-1_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive /var/cache/apt/archives/linux-image-surface_6.0.12-surface-1_amd64.deb (--unpack):
 dpkg-deb --control subprocess returned error exit status 2
Errors were encountered while processing:
 /var/cache/apt/archives/linux-headers-6.0.12-surface_6.0.12-surface-1_amd64.deb
 /var/cache/apt/archives/linux-headers-surface_6.0.12-surface-1_amd64.deb
 /var/cache/apt/archives/linux-image-6.0.12-surface_6.0.12-surface-1_amd64.deb
 /var/cache/apt/archives/linux-image-surface_6.0.12-surface-1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

### Environment

- Hardware model: Surface Pro 6 (i7)
- Kernel version: `Linux mint-box 6.0.9-surface #1 SMP PREEMPT_DYNAMIC Thu Nov 24 19:45:07 UTC 2022 x86_64 GNU/Linux`
- Distribution: Linux Mint Debian Edition 5 ("Elsie")
- `dmesg` output: [dmesg.txt](https://github.com/linux-surface/linux-surface/files/10198984/dmesg.txt)

Any advice?

Thanks,
kingorgg

**Comment by zjosua** [comment]
Same issue here.

Debian's dpkg [doesn't support zstd](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892664).
Implementation is being discussed in the bug report thread and [here](https://lists.debian.org/debian-dpkg/2022/07/msg00006.html) (They use mailing lists. In 2022. I can't even...).

Anyway, I suggest that you keep uploading the Debian packages with tar compression or whatever was used until the last release.

**Comment by althk** [comment]
It looks like since `dpkg 1.19.0.5ubuntu2` the default compression is zstd. and between v6.0.11 and v6.0.12, there was update to the build container to use Ubuntu 22.04 ([this commit](https://github.com/linux-surface/linux-surface/commit/0da8ac628656cc72821b4bc1a636f089a8e5c4fa)) which probably started using this newer dpkg version (my guess without comparing the dpkg versions of 20.04 and 22.04 directly).

the build script will need to be updated to pass `--Zxz` (which was the previous default)

**Comment by qzed** [comment]
> It looks like since `dpkg 1.19.0.5ubuntu2` the default compression is zstd. and between v6.0.11 and v6.0.12, there was update to the build container to use Ubuntu 22.04 ([this commit](https://github.com/linux-surface/linux-surface/commit/0da8ac628656cc72821b4bc1a636f089a8e5c4fa)) which probably started using this newer dpkg version (my guess without comparing the dpkg versions of 20.04 and 22.04 directly).

Correct. Had to update the build container due to some changes in how the Ubuntu sources we use handle configs. And due to the update some defaults got changed. Will try to sort that out now. Sorry for the inconvenience.

**Comment by qzed** [comment]
Okay, should hopefully be fixed in e7847fd836bdbcbc81e4d8deca6dcaaa42dba31d. New packages are being built right now.

**Comment by kwai-chang** [comment]
Same to my, already new debian 11.5 install but now see already in building. Thought because of different installing maybe anywhere defective package

**Comment by althk** [comment]
confirming that v6.0.12-surface-3 installed as expected on Debian testing, thanks @qzed 

```
zsh 3594 % uname -rv
6.0.12-surface #3 SMP PREEMPT_DYNAMIC Sun Dec 11 00:34:45 UTC 2022
```

**Comment by bouleetbil34** [comment]
Same error on my surface pro6 with linux-image-surface_6.0.12-surface-3_amd64.deb with debian bookworm
`dpkg-deb: error: archive '/var/cache/apt/archives/linux-headers-surface_6.0.12-surface-3_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up`

**Comment by StollD** [comment]
The compression was only changed for the kernel package, not for the kernel meta package that provides the name `linux-image-surface`. A new version is building right now, in the meantime you can probably work around this by installing `linux-image-6.0.12-surface` directly.

**Comment by qzed** [comment]
The repo should now have some new working kernel packages.

**Comment by bouleetbil34** [comment]
Same error with the latest package
`dpkg-deb: error: archive '/var/cache/apt/archives/linux-headers-surface_6.0.12-surface-4_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up`

**Comment by qzed** [comment]
Turns out that passing `-Zxz` to `dpkg-buildpackage` only sets the compression of the source package... It looks like we need to override `dh_builddeb` in `debian/rules` for it to stick. Next try is building...

**Comment by StollD** [comment]
Would building the kernel on Debian 11 maybe make things easier?

**Comment by qzed** [comment]
Probably. We went with an Ubuntu container as the sources are based on Ubuntu. Anyway, it should work now. The .deb files now contain `.tar.xz` files as expected. But if there are more problems we should probably consider switching to Debian 11.

**Comment by bouleetbil34** [comment]
fixed for me with 6.0.12-surface-5 with debian bookworm

**Comment by natenasby** [comment]
Thanks so much. 6.0.12-surface now works on Debian Bullseye.
